File: atbash/atbash.scala
object Atbash {
  def atbash(ch : Char) : Char = {
      ch match {
        case it if it.isUpper => ('Z' - ch + 'A').toChar
        case it if it.isLower => ('z' - ch + 'a').toChar 
        case _                => ch
      }
  }

  def main(args : Array[String]):Unit = {
    println(s"Arg was ${args(0)}")
    println(args(0) map atbash)
  }
}

File: atbash/atbash.fs
// a version in FSharp

let atbash ch =
  let chint = int ch
  match ch with
   | _ when System.Char.IsUpper ch -> char (int 'Z' - chint + int 'A')
   | _ when System.Char.IsLower ch -> char (int 'z' - chint + int 'a')
   | _                             -> ch

[<EntryPoint>]
let main argv =
  let str = if argv.Length > 0 then argv.[0] else "wizard"
  printfn "%s\n%s" str (String.map atbash str)
  0
